model/host/host.go

```diff
@@ -1232,32 +1223,32 @@ func getNumNewParentsAndHostsToSpawn(pool *evergreen.ContainerPool, newHostsNeed
 	}
 
 	if !ignoreMaxHosts { // only want to spawn amount of parents allowed based on pool size
-		if numNewParentsToSpawn, err = parentCapacity(parentDistro, numNewParentsToSpawn, len(currentParents), pool); err != nil {
+		if numNewParentsToSpawn, err = parentCapacity(parentDistro, numNewParentsToSpawn, len(existingParents), pool); err != nil {
 			return 0, 0, errors.Wrap(err, "could not calculate number of parents needed to spawn")
 		}
 	}
 
 	// only want to spawn amount of containers we can fit on running/uninitialized parents
-	newHostsNeeded = containerCapacity(len(currentParents)+numNewParentsToSpawn, len(existingContainers), newHostsNeeded, pool.MaxContainers)
+	newHostsNeeded = containerCapacity(len(existingParents)+numNewParentsToSpawn, len(existingContainers), newHostsNeeded, pool.MaxContainers)
 	return numNewParentsToSpawn, newHostsNeeded, nil
 }
 
 // numNewParentsNeeded returns the number of additional parents needed to
 // accommodate new containers
 func numNewParentsNeeded(params newParentsNeededParams) int {
-	numPossibleContainers := params.numUphostParents * params.maxContainers
-	numPossibleContainersNeeded := params.numExistingContainers + params.numContainersNeeded
-	numAvailableContainers := numPossibleContainers - params.numExistingContainers
+	existingCapacity := params.numExistingParents * params.maxContainers
+	newCapacityNeeded := params.numContainersNeeded - (existingCapacity - params.numExistingContainers)
```

Comment:
So `existingContainers` comes from `existingParents`, which only includes parents with an uphost status, i.e. not decommissioned. It will include containers that have been assigned to uninitialized parents however.